c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,453 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
+      TiRuntime runtime = ti_create_runtime(arch);
+      for (int i = 0; i < 4; ++i) {
+        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT << i;
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        TI_ASSERT(memory != TI_NULL_HANDLE);
+      }
+
+      // Attempt to run out of the memory
+      {
+        allocate_info.size = 1000000000000000000;
+        ti_allocate_memory(runtime, &allocate_info);
+        error = ti_get_last_error(0, nullptr);
+        TI_ASSERT(error == TI_ERROR_OUT_OF_MEMORY);
+        allocate_info.size = 1024;
+      }
+
+      // runtime and allocate_info are both null
+      {
+        ti_allocate_memory(TI_NULL_HANDLE, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is not null, allocate_info is null
+      {
+        ti_allocate_memory(runtime, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null, allocate is not null;
+      {
+        ti_allocate_memory(TI_NULL_HANDLE, &allocate_info);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorFreeMemory) {
+  auto inner = [](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      TiRuntime runtime = ti_create_runtime(arch);
+      TiMemoryAllocateInfo *allocate_info = new TiMemoryAllocateInfo;
+      allocate_info->size = 1024;
+      allocate_info->usage = TI_MEMORY_USAGE_STORAGE_BIT;
+      TiMemory memory = ti_allocate_memory(runtime, allocate_info);
+      ti_free_memory(runtime, memory);
+      CHECK_TAICHI_SUCCESS();
+
+      // runtime & allocate_info are both null
+      {
+        ti_free_memory(TI_NULL_HANDLE, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null and allocate_info is valid
+      {
+        TiMemoryAllocateInfo *allocate_info = new TiMemoryAllocateInfo;
+        allocate_info->size = 1024;
+        allocate_info->usage = TI_MEMORY_USAGE_STORAGE_BIT;
+        TiMemory memory = ti_allocate_memory(runtime, allocate_info);
+        ti_free_memory(TI_NULL_HANDLE, memory);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is not null and allocate_info is null
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        ti_free_memory(runtime, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorMapMemory) {
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      TiRuntime runtime = ti_create_runtime(arch);
+      TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+      ti_map_memory(runtime, memory);
+      CHECK_TAICHI_SUCCESS();
```

Comment:
Same.

Reply:
Also you should check whether the mapped pointer is non-null; and remember to unmap the memory.